app/scripts/modules/core/src/presentation/tables/standardGridTableLayout.less

```diff
@@ -0,0 +1,196 @@
+.StandardGridTableLayout {
+  display: grid;
+
+  .standard-grid-table-header {
+    display: grid;
+    position: sticky;
+    top: 0;
+    font-weight: 600;
+    -webkit-font-smoothing: antialiased;
+    -moz-osx-font-smoothing: grayscale;
+    color: var(--color-regal);
+    font-size: 13px;
+    z-index: 100;
+    align-items: flex-end;
+    min-height: 48px;
+    grid-column: 1 / -1;
+    background-color: var(--color-smoke);
+    border-bottom: 2px solid var(--color-blumine);
+
+    .standard-grid-table-cell {
+      align-items: flex-end;
+      padding-bottom: 6px;
+
+      &.sortable {
+        border-radius: 16px 16px 0 0;
+        cursor: pointer;
+        height: 32px;
+
+        &:hover {
+          background-color: rgba(160, 180, 220, 0.15);
+        }
+
+        & > .sort-indicator {
+          border-radius: 16px;
+          background-color: var(--color-blumine);
+          width: 16px;
+          height: 16px;
+          margin-left: 6px;
+          margin-bottom: 1px;
+          position: relative;
+
+          &:after {
+            content: '';
+            position: absolute;
+            top: 50%;
+            left: 50%;
+            width: 0;
+            height: 0;
+            margin-left: -4px;
+            border-left: 4px solid transparent;
+            border-right: 4px solid transparent;
+          }
+
+          &.ascending:after {
+            margin-top: -2px;
+            border-bottom: 4px solid var(--color-smoke);
+          }
+
+          &.descending:after {
+            margin-top: -1px;
+            border-top: 4px solid var(--color-smoke);
+          }
+        }
+      }
+
+      &.header-expand-toggle {
+        display: flex;
+        justify-content: center;
+        align-items: flex-end;
+        cursor: pointer;
+        padding-left: 0;
+        border-radius: 16px 16px 0 0;
+        height: 32px;
+
+        &:hover {
+          background-color: rgba(160, 180, 220, 0.15);
+        }
+
+        & > i {
+          font-size: 24px;
+          margin-bottom: -4px;
+        }
+      }
+    }
+  }
+
+  .standard-grid-table-row {
```

Comment:
Definitely a valid thing to ask! With the caveat that this is all early stages and will probably change a lot, @gcomstock and I have put some time and research into exactly this question and we feel relatively confident that it will be able to serve the majority of the tabular display use cases we're currently aware of. A couple nuances though:

- **Greg and I do not currently see things like infrastructure views, functions, etc. as belonging in the "Table" family.** Greg's perspective is that those views are not really tables, they don't have columns, and it's not the right fit to try to force them into a tabular data API (e.g. something focused on columns and rows). He and I have been calling the table-esque bits of those views "Object lists" or "Object rows", and I suspect that when it comes time to implement views using that visual pattern we will come up with a standalone API whose whole job is to be **really great** at modeling that pattern, since it's one we know will be an important and central building block of our most critical views.

- **The list of places we actually use tabular data is not that long.** I don't think we have a huge diversity of tables today, though Greg is certainly leveraging them more than we have historically. We audited our current and planned table usage and right now it's seeming like this 'standard' layout style will serve:
  - MD history (this will be the first usage to prove it out)
  - Tasks view
  - Kayenta reports table
  - FP rollouts (Netflix internal only)
  - Astrid artifact/server group usage breakdowns (Netflix internal only)

Happy to dig into this a bit more tactically if you're interested, I'm trying to lean into experimenting with this vs. having everything lined up in the first pass.

Reply:
I approve of this message